### PR TITLE
Show the LoaderDialog when syncing the Location data

### PR DESCRIPTION
**Pull request by qaziabubakar-vd**
Show the LoaderDialog when syncing the Location data
Update the LoaderDialog:
    - Make box and progressBar size dynamic
    - Make the progressDialog message as optional
    - Add the ability to show either the ProgressBar or LineSpinFadeLoaderProgressIndicator

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes [https://github.com/opensrp/fhircore/issues/3544]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderDialog.kt [hunk 1]
```diff
@@ -52,58 +52,113 @@ const val LOADER_DIALOG_PROGRESS_MSG_TAG = "loaderDialogProgressMsgTag"
...
-                    text = stringResource(id = R.string.percentage_progress, currentPercentage),
-                    modifier = modifier.padding(horizontal = 3.dp, vertical = 16.dp),
-                  )
-                }
-              }
+            dialogMessage?.let {
+              Text(
+                text = it,
+                color = Color.White,
+                fontSize = 14.sp,
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderDialog.kt:150]: @ellykits @qaziabubakar-vd Does this font size align with the default font size in other areas of the app?

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderDialog.kt [hunk 2]
```diff
@@ -52,58 +52,113 @@ const val LOADER_DIALOG_PROGRESS_MSG_TAG = "loaderDialogProgressMsgTag"
...
+                modifier = modifier.padding(top = 8.dp),
+              )
+            }
+
+            if (showPercentageProgress) {
+              Text(
+                fontSize = 15.sp,
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderDialog.kt:157]: @ellykits @qaziabubakar-vd Does this font size align with the default font size in other areas of the app?

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderDialog.kt (hunk 2 above)
Comment (qaziabubakar-vd) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderDialog.kt:157]: I haven't changed any font sizes, we were showing this same font size before in the LoaderDialog class, I just refactored it a little for my implementation. 
I can confirm with @ellykits if the size is aligned with the rest of the app.

**Comment by qaziabubakar-vd** [comment]
Here is the video recording of the implementation for showing the LoaderDialog: 
https://github.com/user-attachments/assets/874b6271-9d1f-4315-ac5a-30967d557375

**Comment by qaziabubakar-vd** [comment]
For showing the LoaderDialog for the CustomSyncWorker, that is still a WIP.

**Summary of Attempted Solutions so far**
- Used `observeForever` with `getWorkInfosForUniqueWorkLiveData`
- Tried observing worker status using observeForever on getWorkInfosForUniqueWorkLiveData.
- The worker status updates were not received in real-time. Instead, all logs related to the worker were delayed and displayed only after the worker finished execution.
- Created a combined extension method `schedulePeriodicallyWithObserver` to both schedule and observe workers.
- Despite ensuring LiveData observation and worker execution were on separate threads, status updates were not reliably delivered in real-time.

**Next Steps, Current flow that I am trying:**

- Replace LiveData-based observation with a Flow-based solution that can emit worker status updates incrementally. This might offer more reliable real-time updates with WorkManager.

- Will focus on tracking the progress of the `fetchResources` function within the worker by adding progress updates during its execution. Specifically, we plan to:

- **Emit Progress from fetchResources:** Introduce a mechanism, such as calling setProgress(workDataOf("progress" to value)), at key points in the fetchResources function:
- After each successful call to fhirResourceDataSource.getResource() or getResourceWithGatewayModeHeader().

- **Observe Progress Updates:**Use WorkManager.getWorkInfoByIdLiveData() to observe these progress updates in real-time. This will allow us to get incremental updates about the resources being fetched and processed.

This approach might give us more control into what the worker is doing at runtime and help identify bottlenecks or delays in the `fetchResources` execution.
cc: @ellykits

**Pull request merged**